src/script/lua_api/l_item.cpp

```diff
@@ -525,6 +525,29 @@ int ModApiItemMod::l_register_item_raw(lua_State *L)
 	return 0; /* number of results */
 }
 
+// unregister_item(name)
+int ModApiItemMod::l_unregister_item_raw(lua_State *L)
+{
+	NO_MAP_LOCK_REQUIRED;
+	std::string name = luaL_checkstring(L, 1);
+
+	// Get the writable item definition manager from the server
+	IWritableItemDefManager *idef =
+			getServer(L)->getWritableItemDefManager();
+
+	// Unregister the node
+	if (idef->get(name).type == ITEM_NODE) {
+		IWritableNodeDefManager *ndef =
+			getServer(L)->getWritableNodeDefManager();
+		ndef->unallocateName(name);
+	}
+
+	// Unregister item
+	idef->unregisterItem(name);
+
+	return 0; /* number of results */
```

Comment:
"Commenting a lot" isn't really a good thing on its own.  Comments need to have purpose, for example, explaining something unclear to a reader where the purpose of a piece of code might not be immediately clear.  If you're just going to repeat what the code itself says, then there's no value added, and instead adds noise to the code that would otherwise be perfectly clean.

In this particular case, it is generally understood from the surrounding context that `return 0;` is part of the interface between Lua and C++.  If the developer reading this code doesn't know what `return 0` does in this context, perhaps that's a sign that they should read the [Programming in [Lua](http://www.lua.org/pil/contents.html) handbook before writing Minetest code for the Lua scripting interface.